### PR TITLE
Fix: scope legacy settings ajax to the target form

### PR DESCRIPTION
**Pull request by nkb-bd**
## What does this PR do and why?

Passes the requested `form_id` into the legacy general-settings and email-notification AJAX permission checks. This ensures a user must have access to the specific form being updated instead of only holding a broad forms capability.

**Related issue:** N/A

## Scope

- [x] Free plugin
- [ ] Pro plugin
- [ ] Both

## Changes

- [x] PHP (backend logic, models, services, hooks)
- [ ] Vue/React (admin UI, block editor)
- [ ] CSS/SCSS (styling)
- [ ] Database (migrations, schema changes)
- [ ] REST API (new or changed endpoints)
- [ ] Build/config (Vite, composer, CI)

## How to test

1. Create two forms and a user who can manage only one of them.
2. Save general settings and email notifications for the allowed form and confirm both requests succeed.
3. Repeat the same requests for a disallowed form by changing the `form_id` in the request.
4. Confirm the unauthorized requests are blocked.

## Anything the reviewer should know?

This only affects the remaining legacy AJAX handlers for form settings. It does not change the payload shape or the saved data.

### CODE REVIEW

**Review by autharif — CHANGES_REQUESTED** [review]
<h3>Summary</h3>
<p>The change is narrowly scoped, but there is one blocker to address before merge.</p>
<p><strong>Key changes:</strong></p>
<ul>
<li>Both legacy settings AJAX handlers now pass <code>form_id</code> into <code>Acl::verify()</code>, so authorization is no longer purely capability-wide.</li>
<li>The change reuses the centralized ACL / form-permission path instead of introducing endpoint-specific auth logic.</li>
</ul>
<h3>Findings</h3>
<p>Must fix before merge.</p>
<ul>
<li>Normalize form_id before ACL (<code>app/Hooks/Ajax.php:59:80</code>)</li>
</ul>
<h3>Confidence Score: 3/5</h3>
<ul>
<li>Merge stance: <code>REQUEST CHANGES</code> with 1 confirmed blocker-level finding(s).</li>
<li>Verification confirmed 0 Critical and 1 Important finding(s) in the changed files. Change improves form scoping, but authorization and the write target are still derived from differently normalized values.</li>
</ul>
<details><summary>Prompt To Fix All With AI</summary>

```text
Apply the confirmed review fixes for fluentform/fluentform#862 at commit 2436b9bb3ccd4b333cfb3fff727ef3acaeb116cf.
Keep behavior unchanged outside these issues.

1. Normalize form_id before ACL
What: The new scope check consumes raw request data. `Request::get(&#39;form_id&#39;)` can yield an array, `Acl::hasFormPermission()` treats arrays as valid form sets, and the downstream settings services cast the same value back to an int. A crafted `form_id[]` can satisfy auth for one allowed form while writing settings under a different form ID.
Why: This leaves the intended per-form authorization bypassable and lets a user with access to one form mutate another form&#39;s settings.
Fix: Normalize `form_id` to a strict scalar integer before calling `Acl::verify()`, reject non-numeric or array values, and pass the same normalized ID into the service payload so auth and mutation target the same form.
```
</details>
<p><sub>Last reviewed commit: <a href="https://github.com/fluentform/fluentform/commit/2436b9bb3ccd4b333cfb3fff727ef3acaeb116cf">Fix: scope legacy settings ajax to the target form</a></sub></p>

**Review by autharif — COMMENTED** [review]
Rechecked [`f2d6742`](https://github.com/fluentform/fluentform/pull/862/commits/f2d67422ed6d43c10c99187e7cf464844977b310). Some blocking findings were cleared. PR still blocked.
<p><strong>Findings</strong></p>
<p>Merge blocked until these are fixed.</p>
<ul>
<li>Report dashboard endpoints now require form_id but UI defaults to All Forms (<code>app/Http/Policies/ReportPolicy.php:28:96</code>)</li>
<li>Report APIs permit dashboard-only access to analytics data (<code>app/Http/Policies/ReportPolicy.php:35</code>)</li>
<li>Submissions report policy blocks existing all-forms chart path (<code>app/Http/Policies/ReportPolicy.php:68:71</code>)</li>
<li>Subscription cancel path bypasses form-scoped ACL for payment managers (<code>app/Modules/Payments/TransactionShortcodes.php:454</code>)</li>
</ul>

**Review by autharif — APPROVED** [review]
Rechecked [`10a0cff`](https://github.com/fluentform/fluentform/pull/862/commits/10a0cff7d7dc3468dd304e535de25f125b9ee579). Blocking findings cleared.
